app/src/main/java/org/fossasia/openevent/app/core/auth/login/LoginFragment.java

```diff
@@ -49,25 +48,60 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
     @Override
     public void onStart() {
         super.onStart();
-        getPresenter().attach(this);
-        binding.setLogin(getPresenter().getLogin());
-        getPresenter().start();
+
+        //attaching the view model
+        loginFragmentViewModel = ViewModelProviders.of(this, viewModelFactory).get(LoginViewModel.class);
+        handleEmails();
+        handleNewIntent();
+
+        //getPresenter().start();
+        binding.setLogin(loginFragmentViewModel.getLogin());
         binding.btnLogin.setOnClickListener(view -> {
             if (!validator.validate())
                 return;
 
             ViewUtils.hideKeyboard(view);
-
             String url = binding.url.baseUrl.getText().toString().trim();
-            getPresenter().setBaseUrl(url, binding.url.overrideUrl.isChecked());
-            getPresenter().login();
+
+            loginFragmentViewModel.setBaseUrl(url, binding.url.overrideUrl.isChecked());
+            loginFragmentViewModel.showProgress().observe(this, response -> {
+                showProgress(response);
+            });
+            loginFragmentViewModel.login().observe(this, this::onResponse);
+
+            handleNewIntent();
         });
 
         binding.signUpLink.setOnClickListener(view -> openSignUpPage());
-
         binding.forgotPasswordLink.setOnClickListener(view -> openForgotPasswordPage());
     }
 
+    public void handleNewIntent() {
+        loginFragmentViewModel.getLoginStatus().observe(this, isLoggedIn -> {
+            if(isLoggedIn) {
+                Intent intent = new Intent(getActivity(), MainActivity.class);
+                intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
+                startActivity(intent);
+                getActivity().finish();
+            }
+        });
+    }
+
+    private void handleEmails() {
+        loginFragmentViewModel.getEmailList().observe(this, emailsList -> {
+                binding.emailDropdown.setAdapter(
+                    new ArrayAdapter<>(getActivity(), android.R.layout.simple_list_item_1, new ArrayList<>(emailsList)));
+        });
+    }
+
+    private void onResponse(Boolean state) {
+        if(!state) {
+            loginFragmentViewModel.showError().observe(this, message -> {
+                showError(message);
+            });
+        }
```

Comment:
Can't understand this logic

Reply:
this meant that if the user was able to login then the login state would be true but if he encountered any problem then he might not have been able to login and so an error would have occured

Reply:
The error should be subscribed unconditionally, not on observation of a livedata